pkg/controller/controller.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package controller
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/controller/command"
+	didexchangecmd "github.com/hyperledger/aries-framework-go/pkg/controller/command/didexchange"
+	messagingcmd "github.com/hyperledger/aries-framework-go/pkg/controller/command/messaging"
+	vdricmd "github.com/hyperledger/aries-framework-go/pkg/controller/command/vdri"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/rest"
+	didexchangerest "github.com/hyperledger/aries-framework-go/pkg/controller/rest/didexchange"
+	messagingrest "github.com/hyperledger/aries-framework-go/pkg/controller/rest/messaging"
+	vdrirest "github.com/hyperledger/aries-framework-go/pkg/controller/rest/vdri"
+	"github.com/hyperledger/aries-framework-go/pkg/controller/webhook"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/context"
+)
+
+type allOpts struct {
+	webhookURLs  []string
+	defaultLabel string
+	autoAccept   bool
+	msgHandler   command.MessageHandler
+}
+
+// Opt represents a controller option.
+type Opt func(opts *allOpts)
+
+// WithWebhookURLs is an option for setting up a webhook dispatcher which will notify clients of events
+func WithWebhookURLs(webhookURLs ...string) Opt {
```

Comment:
How does this work for the JavaScript (it should get events directly rather than Webhooks.

Reply:
This one is used in case of http notifier, JS version will use callback func version.

aries-agent-rest gets webhook from cli, for javascript it should be in wasm initialization params, by default I am thinking of javascript callback.

Reply:
Take a look at generators.
https://modernweb.com/replacing-callbacks-with-es6-generators/

Reply:
async generators: http://thecodebarbarian.com/async-generator-functions-in-javascript.html